src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -80,7 +80,21 @@ public virtual async Task InvokeAsync(HttpContext context, RequestDelegate next)
                         GraphQLRequest[] deserializationResult;
                         try
                         {
+#if NET5_0_OR_GREATER
+                            var sourceEncoding = GetEncoding(httpRequest);
+                            // Wrap content stream into a transcoding stream that buffers the data transcoded from the sourceEncoding to utf-8.
+                            if (sourceEncoding != null && sourceEncoding != System.Text.Encoding.UTF8)
+                            {
+                                using var tempStream = System.Text.Encoding.CreateTranscodingStream(httpRequest.Body, innerStreamEncoding: sourceEncoding, outerStreamEncoding: System.Text.Encoding.UTF8);
+                                deserializationResult = await _serializer.ReadAsync<GraphQLRequest[]>(tempStream, cancellationToken);
+                            }
+                            else
+                            {
+                                deserializationResult = await _serializer.ReadAsync<GraphQLRequest[]>(httpRequest.Body, cancellationToken);
+                            }
+#else
                             deserializationResult = await _serializer.ReadAsync<GraphQLRequest[]>(httpRequest.Body, cancellationToken);
```

Comment:
It would be nice to deduplicate this line of code which now appears 3 times... but it's okay if that's not practical.

Reply:
Counteroffer  - drop 3.1 tfm for v7 .

Reply:
No, supporting 3.1 is essential.  I have projects that use 3.1 and cannot be upgraded at this time.  I'm sure others do as well.

Reply:
3.1 server or 3.1 GraphQL.NET ?

Reply:
.NET Core 3.1